jetstream/cli.py

```diff
@@ -94,8 +108,15 @@ def execute(
         if configuration_map is not None:
             raise Exception("Custom configurations are not supported when running with Argo")
 
-        experiments_config = [
-            {"date": date.strftime("%Y-%m-%d"), "slug": slug} for (slug, date) in worklist
+        experiments_config: Dict[str, List[str]] = {}
+        for (slug, date) in worklist:
+            if slug not in experiments_config:
+                experiments_config[slug] = []
+
+            experiments_config[slug].append(date.strftime("%Y-%m-%d"))
```

Comment:
This can be more concise, if not necessarily more legible: `experiments_config.setdefault(slug, []).append(date.strftime("%Y-%m-%d"))`